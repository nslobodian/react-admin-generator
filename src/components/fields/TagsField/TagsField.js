import React from 'react'
import s from './TagsField.scss'
import Dropdown from 'components/Dropdown/DropdownElement'
import ManageTagsModal from './ManageTagsModal/index'
import Field from 'redux-form/es/Field'

export const TagsField = ({ showManageTagsModal, notificationTags, name = 'tags' }) => (
  <div className='form-group'>
    <div className='text-muted clearfix'>
      <div className='pull-left'>Tags</div>
      <div className='pull-right'>
        manage tags <span className={s.manageTagsLink} onClick={showManageTagsModal}>here</span>
      </div>
    </div>
    <Field
      component={Dropdown}
      name={name}
      labelKey='name'
      valueKey='notificationTagId'
      multi
      options={notificationTags}
    />
    <ManageTagsModal />
  </div>
)

export default TagsField
