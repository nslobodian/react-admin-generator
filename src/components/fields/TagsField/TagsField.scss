.manageTagsLink {
  text-decoration: underline;
  cursor: pointer;
}