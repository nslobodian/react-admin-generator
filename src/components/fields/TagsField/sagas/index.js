import createNotificationTags from './createNotificationTags'
import deleteNotificationTags from './deleteNotificationTags'
import updateNotificationTags from './updateNotificationTags'

export {
  createNotificationTags,
  deleteNotificationTags,
  updateNotificationTags,
}
