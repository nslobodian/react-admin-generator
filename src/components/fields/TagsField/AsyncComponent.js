import React, { Component } from 'react'
import connect from 'react-redux/es/connect/connect'
import { show } from 'redux-modal/lib/actions'
import { getNotificationTagsAttempt } from 'redux/NotificationTag'
import TagsField from './TagsField'
import { injectSagas } from 'store/sagas'
import { getStore } from 'store/createStore'

export class CreateNewAudienceWrapper extends Component {
  componentWillMount () {
    this.props.getNotificationTagsAttempt()
  }

  componentDidMount () {
    import(/* webpackChunkName: 'Component Tags' */ './sagas')
      .then(({
        createNotificationTags,
        deleteNotificationTags,
        updateNotificationTags,
      }) => {
        injectSagas(getStore(), {
          key: 'CreateNotification',
          sagas: [
            createNotificationTags,
            deleteNotificationTags,
            updateNotificationTags,
          ],
        })
      })
  }

  render () {
    return <TagsField {...this.props} />
  }
}

const mapActionCreators = {
  getNotificationTagsAttempt,
  showManageTagsModal: () => show('mangeTagsModal'),
}

export default connect(({ NotificationTag: { notificationTags } }) => ({ notificationTags }), mapActionCreators)(CreateNewAudienceWrapper)

