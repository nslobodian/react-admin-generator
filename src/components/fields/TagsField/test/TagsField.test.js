import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import TagsField from '../TagsField'

describe('(Components) TagsField', () => {
  it('Should render component properly', () => {
    const props = {
      notificationTags: [],
      showManageTagsModal: jest.fn(),
    }
    const wrapper = shallow(
      <TagsField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
