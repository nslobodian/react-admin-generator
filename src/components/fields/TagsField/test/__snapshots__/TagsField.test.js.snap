// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) TagsField Should render component properly 1`] = `
<div
  className="form-group"
>
  <div
    className="text-muted clearfix"
  >
    <div
      className="pull-left"
    >
      Tags
    </div>
    <div
      className="pull-right"
    >
      manage tags 
      <span
        className="manageTagsLink"
        onClick={[Function]}
      >
        here
      </span>
    </div>
  </div>
  <Field
    component={[Function]}
    labelKey="name"
    multi={true}
    name="tags"
    options={Array []}
    valueKey="notificationTagId"
  />
  <Connect(Connect(ConnectModal(SelectProductModalWrapper))) />
</div>
`;
